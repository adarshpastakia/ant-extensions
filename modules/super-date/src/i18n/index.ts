// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import i18next from "i18next";
import en from "./en.json";
import ar from "./ar.json";
import { I18nKey } from "../utils/i18nKey";

const callback = () => {
  i18next.addResourceBundle("en", I18nKey, en, true, false);
  i18next.addResourceBundle("ar", I18nKey, ar, true, false);
};

if (i18next.isInitialized) {
  callback();
} else {
  i18next.on("initialized", callback);
}
