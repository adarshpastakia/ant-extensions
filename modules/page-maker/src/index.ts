// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import "./i18n";

export { EnumTypes } from "./utils/types";
export type { PageConfig, PageItem, WidgetObject } from "./utils/types";
export { PageMaker } from "./components/PageMaker";
