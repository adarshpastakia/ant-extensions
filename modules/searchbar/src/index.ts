// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT
import "./i18n";

export {Type, Operator} from "./utils/types";
export type { IFilterObject, IFilterField } from "./utils/types";
export { SearchBar } from "./components/SearchBar";
