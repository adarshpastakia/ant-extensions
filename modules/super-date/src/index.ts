// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT
import "./i18n";

export { InlineDatePicker } from "./components/InlineDatePicker";
export { InlineRangePicker } from "./components/InlineRangePicker";
export { RelativeDatePicker } from "./components/RelativeDatePicker";
export { RelativeRangePicker } from "./components/RelativeRangePicker";
export { DateUtils } from "./utils/dateUtils";
export { DateParts } from "./utils/types";
