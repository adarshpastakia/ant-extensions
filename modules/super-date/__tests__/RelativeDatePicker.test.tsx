// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2020
// @license   : MIT

import { render } from "@testing-library/react";
import expect from "expect";
import i18next from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";
import { RelativeDatePicker } from "../src";

i18next
  // pass the i18n instance to react-i18next.
  .use(initReactI18next)
  // init i18next
  // for all options read: https://www.i18next.com/overview/configuration-options
  .init({
    resources: {
      en: {}
    },
    defaultNS: "core",
    fallbackLng: ["en"],
    keySeparator: ".",

    interpolation: {
      escapeValue: false // not needed for react as it escapes by default
    }
  });
i18next.languages = ["en", "ar"];

describe("RelativeDatePicker", () => {
  const { container, rerender, unmount } = render(
    <I18nextProvider i18n={i18next}>
      <RelativeDatePicker value="$now" />
    </I18nextProvider>
  );

  afterAll(unmount);

  it("should render", (done) => {
    expect(container).toMatchSnapshot();
    done();
  });

  it("should change value", (done) => {
    rerender(<RelativeDatePicker value="$week" />);
    expect(container).toMatchSnapshot();
    done();
  });

  it.todo("should fire change event");
});
