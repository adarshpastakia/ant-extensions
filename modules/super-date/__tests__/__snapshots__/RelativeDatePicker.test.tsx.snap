// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelativeDatePicker should change value 1`] = `
<div>
  <div
    class="ant-ext-sd__input"
  >
    <span
      class="ant-input-affix-wrapper"
    >
      <input
        class="ant-input"
        readonly=""
        type="text"
        value="This week"
      />
      <span
        class="ant-input-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`RelativeDatePicker should render 1`] = `
<div>
  <div
    class="ant-ext-sd__input"
  >
    <span
      class="ant-input-affix-wrapper"
    >
      <input
        class="ant-input"
        readonly=""
        type="text"
        value="Now"
      />
      <span
        class="ant-input-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </span>
  </div>
</div>
`;
